### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by ranocha**
I've updated the tests to the standard MATLAB unit test framework.

### CODE REVIEW

**Comment by ketch** [comment]
Thanks, @ranocha!  I think we need to add CVX to the list of dependencies, and also mention that the tests won't run without the global optimization toolbox (even though it is listed as an optional dependency).  Or maybe it's simpler to just list that toolbox as a required dependency.

I checked out this branch and ran the tests.  I am getting two test failures.  Are they all passing for you?

Specifically, the assertion on line 34 of test_polyopt fails, and in TestDiskMonomial I get an error from CVX saying "Your objective function is not a scalar".  I'm guessing these are caused by some changes in CVX.  I'm using Version 2.2 build 1148 (latest stable).

**Comment by ranocha** [comment]
I just checked again - all tests pass for me. 

<details>
<summary>

Specifically, I get the following output.
</summary>

```
ans =

  6×6 table

                      Name                       Passed    Failed    Incomplete    Duration      Details   
    _________________________________________    ______    ______    __________    ________    ____________

    'test_polyopt/TestRealAxisMonomial'          true      false       false        4.3055     [1×1 struct]
    'test_polyopt/TestRealAxisChebyshev'         true      false       false          1.75     [1×1 struct]
    'test_polyopt/TestImaginalyAxisMonomial'     true      false       false         3.203     [1×1 struct]
    'test_polyopt/TestImaginalyAxisChebyshev'    true      false       false        3.6316     [1×1 struct]
    'test_polyopt/TestDiskMonomial'              true      false       false        2.7833     [1×1 struct]
    'test_polyopt/TestDiskBinomial'              true      false       false        4.4747     [1×1 struct]


ans =

  18×6 table

                        Name                         Passed    Failed    Incomplete    Duration      Details   
    _____________________________________________    ______    ______    __________    ________    ____________

    'test_rkopt/TestSSP32'                           true      false       false         1.3515    [1×1 struct]
    'test_rkopt/TestSSP43'                           true      false       false        0.22398    [1×1 struct]
    'test_rkopt/TestSDIRKSSP22'                      true      false       false        0.12856    [1×1 struct]
    'test_rkopt/TestDIRKSSP22'                       true      false       false        0.15223    [1×1 struct]
    'test_rkopt/TestRK22Acc'                         true      false       false        0.21795    [1×1 struct]
    'test_rkopt/TestSSPTSRK2'                        true      false       false         0.3239    [1×1 struct]
    'test_rkopt/TestSSP32Multistart'                 true      false       false       0.049824    [1×1 struct]
    'test_rkopt/TestSSP32MultistartParallel'         true      false       false         13.469    [1×1 struct]
    'test_rkopt/TestSSP43Multistart'                 true      false       false         0.2038    [1×1 struct]
    'test_rkopt/TestSSP43MultistartParallel'         true      false       false          9.261    [1×1 struct]
    'test_rkopt/TestSDIRKSSP22Multistart'            true      false       false       0.090743    [1×1 struct]
    'test_rkopt/TestSDIRKSSP22MultistartParallel'    true      false       false         9.0936    [1×1 struct]
    'test_rkopt/TestDIRKSSP22Multistart'             true      false       false        0.12162    [1×1 struct]
    'test_rkopt/TestDIRKSSP22MultistartParallel'     true      false       false         8.8557    [1×1 struct]
    'test_rkopt/TestRK22AccMultistart'               true      false       false        0.11993    [1×1 struct]
    'test_rkopt/TestRK22AccMultistartParallel'       true      false       false         9.3472    [1×1 struct]
    'test_rkopt/TestRK323SAcc'                       true      false       false         2.8272    [1×1 struct]
    'test_rkopt/TestEmbeddedRK4353Ss'                true      false       false       0.014644    [1×1 struct]
```
</details>

I'm using the same version of CVX:
```
>> cvx_version

---------------------------------------------------------------------------
CVX: Software for Disciplined Convex Programming       (c)2014 CVX Research
Version 2.2, Build 1148 (62bfcca)                  Tue Jan 28 00:51:35 2020
---------------------------------------------------------------------------
```

Which version of MATLAB do you use, @ketch?

File: polyopt/test_polyopt.m [hunk 1]
```diff
@@ -1,48 +1,49 @@
...
+p = 1;
+[h, poly_coeff] = opt_poly_bisect(lambda, s, p, 'monomial');
+assert(all(ismembertol(h, s-1, tol, 'ByRows', true)))
 
-function test_imagaxis_monomial(tol)
-    lamda = spectrum('imagaxis',100);
-    s=randi(5); p=1;
-    [h,poly_coeff]=opt_poly_bisect(lamda,s,p,'monomial')
-    assertElementsAlmostEqual(h,s-1,'relative',tol);
+%% Test imaginaly axis Chebyshev
```
Comment (ketch) [inline polyopt/test_polyopt.m:28]: imaginary

File: polyopt/test_polyopt.m [hunk 2]
```diff
@@ -1,48 +1,49 @@
...
+p = 1;
+[h, poly_coeff] = opt_poly_bisect(lambda, s, p, 'chebyshev');
+assert(all(ismembertol(h, 2*s^2, tol, 'ByRows', true)))
 
-function test_realaxis_chebyshev(tol)
-    lamda = spectrum('realaxis',100);
-    s=randi(10); p=1;
-    [h,poly_coeff]=opt_poly_bisect(lamda,s,p,'chebyshev')
-    assertElementsAlmostEqual(h,2*s^2,'relative',tol);
+%% Test imaginaly axis monomial
```
Comment (ketch) [inline polyopt/test_polyopt.m:21]: imaginary

**Comment by ketch** [comment]
I'm on MATLAB R2018b.  Are you using something newer?

File: polyopt/test_polyopt.m (hunk 2 above)
Comment (ranocha) [reply polyopt/test_polyopt.m:21]: I've fixed this typo in https://github.com/ranocha/RK-opt/commit/08f2c532d462b49cd1d080560ced74d8d52a371d, but the new commit does again not show up here... GitHub is really weird today.

**Comment by ranocha** [comment]
No, even older = MATLAB R2018a...

**Comment by ketch** [comment]
I guess we should test with the latest MATLAB.

**Comment by ranocha** [comment]
Which solver do you use?

**Pull request merged**